pkg/collector/corechecks/snmp/internal/report/report_device_metadata.go

```diff
@@ -238,9 +242,37 @@ func buildNetworkDeviceMetadata(deviceID string, idTags []string, config *checkc
 		OsHostname:     osHostname,
 		DeviceType:     deviceType,
 		Integration:    common.SnmpIntegrationName,
+		RDNSHostname:   hostname,
 	}
 }
 
+func lookupHostnameWithRDNS(ip string) string {
```

Comment:
suggestion: Maybe just move this whole thing over to the rdsquerier as a synchronous interface